frontend/src/components/ResourceNameTooltip.tsx

```diff
@@ -9,52 +9,87 @@ import {
   StackItem,
   Tooltip,
 } from '@patternfly/react-core';
-import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+import { CopyIcon, InfoAltIcon, OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
 import { K8sResourceCommon } from '@openshift/dynamic-plugin-sdk-utils';
 import '~/pages/notebookController/NotebookController.scss';
+import { EitherNotBoth } from '~/typeHelpers';
 
-type ResourceNameTooltipProps = {
-  resource: K8sResourceCommon;
-  children: React.ReactNode;
+type ResourceNameTooltipProps = EitherNotBoth<
+  { resource: K8sResourceCommon },
+  { resourceName: string; resourceKind: string }
+> & {
+  children?: React.ReactNode;
 };
 
-const ResourceNameTooltip: React.FC<ResourceNameTooltipProps> = ({ children, resource }) => (
-  <div>
-    {children}{' '}
-    {resource.metadata?.name && (
-      <div style={{ display: 'inline-block' }}>
-        <Tooltip
-          removeFindDomNode
-          position="right"
-          content={
-            <Stack hasGutter>
-              <StackItem>
-                Resource names and types are used to find your resources in OpenShift.
-              </StackItem>
-              <StackItem>
-                <DescriptionList isCompact isHorizontal>
-                  <DescriptionListGroup>
-                    <DescriptionListTerm>Resource name</DescriptionListTerm>
-                    <DescriptionListDescription>
-                      {resource.metadata.name}
-                    </DescriptionListDescription>
-                  </DescriptionListGroup>
-                  <DescriptionListGroup>
-                    <DescriptionListTerm>Resource type</DescriptionListTerm>
-                    <DescriptionListDescription>{resource.kind}</DescriptionListDescription>
-                  </DescriptionListGroup>
-                </DescriptionList>
-              </StackItem>
-            </Stack>
-          }
-        >
-          <Icon isInline aria-label="More info" role="button" tabIndex={0}>
-            <OutlinedQuestionCircleIcon />
-          </Icon>
-        </Tooltip>
-      </div>
-    )}
-  </div>
-);
+const ResourceNameTooltip: React.FC<ResourceNameTooltipProps> = ({
+  children,
+  resource,
+  resourceName,
+  resourceKind,
+}) => {
+  const [isHovered, setIsHovered] = React.useState(false);
+
+  const copyToClipboard = async (text: string) => {
+    navigator.clipboard.writeText(text);
+  };
+
+  const name = resourceName ?? resource.metadata?.name;
+  const kind = resourceKind ?? resource.kind;
```

Comment:
```suggestion
  const name = resourceName ?? resource?.metadata.name;
  const kind = resourceKind ?? resource?.kind;
```
`resource` can be undefined -- not sure `metadata` ever will be in our use-case

Reply:
this is resolved by removing the option to manually give resourceName and resourceKind